src/main/java/ch/admin/bag/covidcertificate/domain/ProphylaxisRepository.java

```diff
@@ -1,10 +1,19 @@
 package ch.admin.bag.covidcertificate.domain;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.UUID;
 
 @Repository
 public interface ProphylaxisRepository extends JpaRepository<Prophylaxis, UUID> {
+    @Query(value = "select p from Prophylaxis p " +
+            "where p.code = :code ")
+    Prophylaxis findByCode(@Param("code") String code);
```

Comment:
jpa does not need a query here - findByCode gets translated natively to the right query (see https://www.bezkoder.com/jpa-repository-query/#JPA_find_by_field)